File: src/OS/runtime/registry.py
from uuid import UUID

from OS.domain.agent import AgentInstance


class AgentRegistry:
    """
    Agent 注册表。
    """

    def __init__(self):
        self._agents: dict[UUID, AgentInstance] = {}

    def register(self, agent: AgentInstance) -> None:
        """
        注册 Agent。
        """

        if agent.id in self._agents:
            raise ValueError(f"已存在agent{agent.id}。")

        self._agents[agent.id] = agent

    def get(self, agent_id: UUID) -> AgentInstance:
        """
        根据 ID 获取 Agent。
        """

        try:
            return self._agents[agent_id]

        except KeyError:
            raise ValueError(f"未找到agent{agent_id}。")

    def remove(self, agent_id: UUID) -> None:
        """
        删除 Agent。
        """

        if agent_id not in self._agents:
            raise ValueError(f"未找到{agent_id}。")

        del self._agents[agent_id]

File: go/src/runtime/registry.go
package runtime

import (
	"agent-runtime-go/src/domain"
	"fmt"
)

// AgentRegistry 用 ID 保存已注册的 Agent。它由 Runtime 的锁保护。
type AgentRegistry struct {
	agents map[domain.ID]*domain.AgentInstance
}

func NewAgentRegistry() *AgentRegistry {
	return &AgentRegistry{agents: make(map[domain.ID]*domain.AgentInstance)}
}

func (r *AgentRegistry) Register(agent *domain.AgentInstance) error {
	if agent == nil {
		return fmt.Errorf("注册 Agent: Agent 不能为空")
	}
	if _, exists := r.agents[agent.ID]; exists {
		return fmt.Errorf("Agent %s 已存在", agent.ID)
	}
	r.agents[agent.ID] = agent
	return nil
}

func (r *AgentRegistry) Get(agentID domain.ID) (*domain.AgentInstance, error) {
	agent, exists := r.agents[agentID]
	if !exists {
		return nil, fmt.Errorf("找不到 Agent %s", agentID)
	}
	return agent, nil
}

func (r *AgentRegistry) Remove(agentID domain.ID) error {
	if _, exists := r.agents[agentID]; !exists {
		return fmt.Errorf("找不到 Agent %s", agentID)
	}
	delete(r.agents, agentID)
	return nil
}
